crates/forge-runner/src/running.rs

```diff
@@ -286,6 +288,12 @@ pub fn run_test_case(
     let block_context = get_context(&forge_runtime).block_context.clone();
     let execution_resources = get_all_execution_resources(forge_runtime);
 
+    let max_steps = if case.max_steps.is_some() {
+        case.max_steps
+    } else {
+        runner_config.max_steps
+    };
```

Comment:
Please use match, it is safer beasue it checks for exhaustiveness.

Reply:
Also please don't resolve it here. Please set the default value inside case in cli layer and make it not optional. This leaks cli logic into runner.